internal/client/api_version.go

```diff
@@ -162,12 +175,11 @@ func EncodeRequest(req *tikvrpc.Request) (*tikvrpc.Request, error) {
 		newReq.Req = &r
 	case tikvrpc.CmdRawChecksum:
 		r := *req.RawChecksum()
-		oriRanges := r.Ranges
-		r.Ranges = make([]*kvrpcpb.KeyRange, 0, len(oriRanges))
-		for i := 0; i < len(oriRanges); i++ {
-			keyRange := kvrpcpb.KeyRange{}
-			keyRange.StartKey, keyRange.EndKey = EncodeV2Range(ModeRaw, oriRanges[i].StartKey, oriRanges[i].EndKey)
-			r.Ranges = append(r.Ranges, &keyRange)
+		ranges := r.Ranges
+		// ranges in req maybe reused when retry, alloc another slice to avoid changing the range.
+		r.Ranges = make([]*kvrpcpb.KeyRange, 0, len(ranges))
+		for i := 0; i < len(ranges); i++ {
```

Comment:
I think @pingyu might want something look like this: 
```go
r.Ranges = EncodeV2KeyRanges(r.Ranges)
```

where `EncodeV2KeyRanges` implemented by `EncodeV2KeyRange`.

Reply:
ok， get the point.